Serial Communication Protocol

HuntaBadday opened the pull request:
A serial protocol for communication between devices.

HuntaBadday committed:
Create serial-protocol.md

HuntaBadday committed:
Update serial-protocol.md

HuntaBadday committed:
Update serial-protocol.md

HuntaBadday committed:
Update serial-protocol.md

HuntaBadday committed:
Update serial-protocol.md

codemaker4 commented:
Looks great, simple and elegant.
Maybe be a bit more specific about how exactly the physical interface looks like?
Like I know it's just a single gate on both sides, but write some stuff down about that. Doesn't have to be long.

codemaker4 commented:
The consortium voted against this proposal, so changes have to be made.

niknal357 commented:
Consider specifying the order in which data bits are transmitted (e.g. least significant bit first or most significant bit first). This removes ambiguity.

endoddo commented:
I agree with what @codemaker4 and @niknal357 wrote.
Also:
- I think it would be useful to specify how a receiver should behave when receiving an invalid packet, even if you want the behaviour to be up to implementation and what happens when the receiver is unable to receive even if it should just ignore the packet.
- The standard should clearly state that packets can also have an arbitrary delay between them which isn't clear at the moment.

QuickWrite commented:
I believe that the standard should be renamed to _Serial Transmission Protocol_ instead of _Serial Communication Protocol_.

This is because the word "_Communication_" implies a bidirectional information exchange. "_Transmission_" on the other hand is just sending information across.

HuntaBadday commented:
5 people in a discord have had a discussion, and we all agreed that communication can be one directional. Where communication *is* any transmission of data.

HuntaBadday committed:
Update serial-protocol.md

Adds interface specification

HuntaBadday committed:
Update serial-protocol.md

Removes colour specification

HuntaBadday committed:
Update serial-protocol.md

Improves labels

codemaker4 commented:
Maybe add like a quick introduction of what the standards is and what it is meant for before beginning with the actual explanation.

HuntaBadday committed:
Update serial-protocol.md

Adds short description of the standard
Adds specification for the first bit of the data transmitted to be the least significant bit.
Adds explanation to the data flow diagram.

QuickWrite commented:
> I believe that the standard should be renamed to _Serial Transmission Protocol_ instead of _Serial Communication Protocol_.
> 
> This is because the word "_Communication_" implies a bidirectional information exchange. "_Transmission_" on the other hand is just sending information across.
~ @QuickWrite 


> 5 people in a discord have had a discussion, and we all agreed that communication can be one directional. Where communication _is_ any transmission of data.
~ @HuntaBadday 

I think that there was a miscommunication. I really didn't like the name as it could also imply a two way communication. I've proposed the word "transmission" as it is more specific whilst the counter argument was that "communication" could also imply a one way communication which is not a really good excuse for me as it does not give any reason why "communication" is superior than "transmission". 

I hope that this clarifies this a little bit more.

codemaker4 commented:
More factual reason: the rest of the description uses (variants of) the word "transmit" a lot, whilst (variants of) the word "communicate" are only in the title, and nowhere else. So transmission is probably a better word.
I must admit that I was amongst the 5 people in the discord VC that said communication was fine, and I take that statement back now. Sorry for this.

HuntaBadday committed:
Update serial-protocol.md

Changes title

HuntaBadday commented:
I have changed the title to Serial Transmission Protocol

codemaker4 commented:
Serial Transmission Protocol Public Vote

The SMLC internal vote has approved of this standard for serial transmission, so now the community should vote. If 70% of the voters approve of this standard over a week, it becomes official.

Vote here:
https://docs.google.com/forms/d/e/1FAIpQLScjwhPxf0rn-jOUqNzIIHyv6-507JFRJrfKGzS_FtBX28MLhw/viewform?usp=sf_link
